test/invariant/fuzz/handlers/HandlerAllo.t.sol

```diff
@@ -33,65 +34,299 @@ contract HandlerAllo is Setup {
+        bytes memory _data,
+        uint256 _msgValue
+    ) public onlyValidPool(_seed) {
+        uint256 _poolId = _getPoolId(_seed);
+        // Register recipient
         targetCall(
             address(allo),
-            0,
-            abi.encodeWithSelector(
-                IAllo.updatePercentFee.selector,
-                _newPercentFee
-            )
+            _msgValue,
+            abi.encodeWithSelector(IAllo.registerRecipient.selector, _poolId, _recipientAddresses, _data)
         );
     }
 
-    function handler_updateBaseFee(uint256 _newBaseFee) public {
-        // Update the base fee - will revert if wrong caller
+    function handler_batchRegisterRecipient(
+        uint256[] memory _seeds,
+        address[][] memory _recipientAddresses,
+        bytes[] memory _data
+    ) public onlyValidPools(_seeds) {
+        uint256[] memory _poolIds = _getPoolsIds(_seeds);
+        // Batch register recipient - will revert if arrays are not of same length
         targetCall(
             address(allo),
             0,
-            abi.encodeWithSelector(IAllo.updateBaseFee.selector, _newBaseFee)
+            abi.encodeWithSelector(IAllo.batchRegisterRecipient.selector, _poolIds, _recipientAddresses, _data)
+        );
+    }
+
+    function handler_fundPool(uint256 _seed, uint256 _amount, uint256 _msgValue) public onlyValidPool(_seed) {
+        uint256 _poolId = _getPoolId(_seed);
+        uint256 _previousBalance = token.balanceOf(address(msg.sender));
+        if (_previousBalance > 0) {
+            _amount = bound(_amount, 0, type(uint256).max - _previousBalance);
+        }
+
+        if (_amount > 0) {
+            FuzzERC20(address(token)).mint(address(msg.sender), _amount);
+        }
+
+        // Fund pool - will revert if the amount is zero or if pool token is native and message value is != amount
+        targetCall(address(allo), _msgValue, abi.encodeWithSelector(IAllo.fundPool.selector, _poolId, _amount));
+    }
+
+    function handler_allocate(
+        uint256 _seed,
+        address[] memory _recipients,
+        uint256[] memory _amounts,
+        bytes memory _data,
+        uint256 _msgValue
+    ) public onlyValidPool(_seed) {
+        uint256 _poolId = _getPoolId(_seed);
+        // Allocate - allocate to a recipient or multiple recipients
+        targetCall(
+            address(allo),
+            _msgValue,
+            abi.encodeWithSelector(IAllo.allocate.selector, _poolId, _recipients, _amounts, _data)
+        );
+    }
+
+    function handler_batchAllocate(
+        uint256[] calldata _seeds,
+        address[][] calldata _recipients,
+        uint256[][] calldata _amounts,
+        uint256[] calldata _values,
+        bytes[] memory _datas,
+        uint256 _msgValue
+    ) public onlyValidPools(_seeds) {
+        uint256[] memory _poolIds = _getPoolsIds(_seeds);
+        // Batch allocate - allocate to multiple pools and recipients, will revert if arrays are not of same length
```

Comment:
I think it will therefore revert a lot (for the array mismatch), should we constrain their length (ie based on the min one, downsizing the other, like `assembly { mstore(array, minLength) }` ?